src/mempool/mempool.rs

```diff
@@ -160,6 +220,55 @@ impl Mempool {
         return true;
     }
 
+    async fn insert(&mut self, message: MempoolMessage) {
+        // TODO(aditi): Maybe we don't need to run validations here?
+        if self.message_is_valid(&message) {
+            let fid = message.fid();
+            let shard_id = self.message_router.route_message(fid, self.num_shards);
+            // TODO(aditi): We need a size limit on the mempool and we need to figure out what to do if it's exceeded
+            match self.messages.get_mut(&shard_id) {
+                None => {
+                    let mut messages = BTreeMap::new();
+                    messages.insert(message.mempool_key(), message.clone());
+                    self.messages.insert(shard_id, messages);
+                    self.statsd_client
+                        .gauge_with_shard(shard_id, "mempool.size", 1);
+                }
+                Some(messages) => {
+                    if messages.len() >= self.capacity_per_shard {
```

Comment:
I think it'd be good to add some logging here so it's easier to debug cases where messages that were successfully submitted didn't make it onto the network. Having fid in a log tag would be useful.

Reply:
```suggestion
                    if messages.len() >= self.capacity_per_shard {
                    warn!(fid = message.fid(), "Message dropped due to mempool being over capacity");
```
Something along these lines?

Reply:
https://github.com/farcasterxyz/snapchain/pull/200/commits/118c035be8bc035f2d262a5cbb23553651a3e8dc

Reply:
Yes, can you add the hash of the message or some identifier too?

Reply:
Done!